src/api/app/models/bs_request_action/differ/for_source.rb

```diff
@@ -50,7 +50,8 @@ def query
       end
 
       def source_diff(project_name, package_name, query)
-        Backend::Api::Sources::Package.source_diff(project_name, package_name, query)
+        diff = Backend::Api::Sources::Package.source_diff(project_name, package_name, query)
+        diff.valid_encoding? ? diff : diff.encode('UTF-8', 'binary', invalid: :replace, undef: :replace)
```

Comment:
Why not doing it directly in: ``Backend::Api::Sources::Package.source_diff``?

Reply:
Hm, I would prefer let the Backend::Api::* just wrap the calls to the backend and process the data somewhere else.
But given that we only use this in the webui, there is no good reason to not add it there.

... I'll update it.